Makefile

```diff
@@ -381,6 +381,16 @@ list-citation:
 		> authors.csv
 	python sort-authors-list.py
 
+## cpp-demos   : run programs demonstrating access to the (unstable) C++ API
+cpp-demos:
```

Comment:
could you make these depend on the shared library build step? Just tried to run it locally on a fresh checkout and you get a "weird" error message. 
`cpp-demos:` -> `cpp-demos: sharedobj` should work I think